### PR TITLE
Widget id validation issue 8

### PR DESCRIPTION
**Pull request by camstuart**
Some minor project updates (editorconfig and gitignore), but mostly to satisfy #8 

The Validation rules for app ID are not exhaustive, but should be more helpful to save some errors when people attempt to build their project and discover that app ID's such as `hello-world` are not valid.

The string comparison should still allow for `*.hello.world` if your Apple App ID is a 'wild card' yee haa!

### CODE REVIEW

File: app/index.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ var projectTypes = [{
 var props;
 
 module.exports = generators.Base.extend({
-  prompting: function() {
+  prompting: function () {
```
Comment (ralfstx) [inline app/index.js]: This commit adds some unnecessary whitespace. Please leave the formatting intact.

**Comment by camstuart** [comment]
Updated as requested. would you be happy to have an `.eslintrc` in the project root? 
I'm a big fan of applying code formatting and standards as part of IDE formatting capabilities

If so, specify the particular standard (eg, airbnb)

**Comment by ralfstx** [comment]
Hi Cam, thanks for the contribution!

So far, we haven't been able to accept Github PRs because of our code review system. I've made some adjustments now and it would be cool if you could help me test it. This would require two changes on your side:

1. Since we have to approve the commits one-by-one, it would be helpful if you could open separate PRs for each commit. You could also combine the config files changes into one, if that makes it easier.

2. We believe in the value of good commit messages. May I ask you to have a look at [1] and apply the "seven rules" to your commits? That would be great!

Thanks in advance,
Ralf

[1] http://chris.beams.io/posts/git-commit/

**Comment by camstuart** [comment]
Ok, will do as two PR's, stay tuned

**Comment by ralfstx** [comment]
As for `.eslintrc`: Absolutely! For the template I think it should be a minimal one, such as https://github.com/eclipsesource/generator-tabris-js/blob/master/.eslintrc, but either without any rules (not to enforce a certain style on new projects) or with a minimal set of rules (such as `"no-trailing-spaces": "error"`) for demonstration.

**Comment by camstuart** [comment]
Ahh, I was thinking more in the code base of this repo for `.eslintrc` to make the coding style for contributions and the acceptance of PR's more streamlined. 

But agreed, a non-opinionated `.eslintrc` and an `.editorconfig` would go nicely in the generated output. I'll add an issue, and come back to that.

**Comment by ralfstx** [comment]
We already have an `.eslintrc` in the repository root, probably not strict enough. Let's copy the one from [tabris-js](https://github.com/eclipsesource/tabris-js/blob/master/.eslintrc).
